File: src/main/java/dev/turingcomplete/intellijbytecodeplugin/openclassfiles/_internal/ClassNameProviderConfigurations.java
package dev.turingcomplete.intellijbytecodeplugin.openclassfiles._internal;

import org.jetbrains.kotlin.idea.debugger.core.ClassNameProvider;

final class ClassNameProviderConfigurations {

  private ClassNameProviderConfigurations() {}

  static ClassNameProvider.Configuration defaultReturningLambdaParentClass() {
    return new ClassNameProvider.Configuration(
        ClassNameProvider.Configuration.Companion.getDEFAULT().getFindInlineUseSites(), true);
  }
}

File: src/main/kotlin/dev/turingcomplete/intellijbytecodeplugin/openclassfiles/_internal/ClassNameProviderConfigurations.kt
package dev.turingcomplete.intellijbytecodeplugin.openclassfiles._internal

import org.jetbrains.kotlin.idea.debugger.core.ClassNameProvider

internal object ClassNameProviderConfigurations {
  fun defaultReturningLambdaParentClass(): ClassNameProvider.Configuration =
    ClassNameProvider.Configuration::class
      .java
      .getDeclaredConstructor(Boolean::class.javaPrimitiveType, Boolean::class.javaPrimitiveType)
      .newInstance(ClassNameProvider.Configuration.DEFAULT.findInlineUseSites, true)
}
